### PR TITLE
#167816359 allow user to access all articles

### PR DESCRIPTION
**Pull request by Ramadhan0**
#### What does this PR do?
 * Enable  users to access all articles
* Enable users to search a certain article
* Enable users to access all pages
#### Description of Task to be completed?
 * Create a reusable component that can be used for ``search``  ``get all article`` and ``get article by category``
#### How should this be manually tested?
* Clone the repo to the local machine.
* Run ``npm install``   to install the dependencies.
* Set up the   ``.env`` as instructed by the   ``.env-example``
* Run  ``npm dev``.`.
*  Navigate to ``/articles``.

### [166240796](https://www.pivotaltracker.com/story/show/166240796)

Screen Shot 2019-09-04 at 21 47 58
Screen Shot 2019-09-04 at 21 48 17
Screen Shot 2019-09-04 at 21 59 37

### CODE REVIEW

File: src/Style/Sass/component/_articles.scss [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+            background: #2F3640;
+
+            &.searchIcon {
+                color: white;
+            }
+        }
+
+    }
+
+    &__displayArticles {
```
Comment (truestbyheart) [inline src/Style/Sass/component/_articles.scss]: Please use small letters only, pascal is not a convention we use in CSS class naming @ramadhan

File: src/Style/Sass/base/_base.scss [hunk 2]
```diff
@@ -26,4 +26,9 @@ nav {
 #root {
   width: 100%;
   overflow-x: hidden;
+
+  #root {
```
Comment (truestbyheart) [inline src/Style/Sass/base/_base.scss]: this  style is useless just remove it and see what happens

**Review by truestbyheart — CHANGES_REQUESTED** [review]

**Review by truestbyheart — APPROVED** [review]

**Review by nkpremices — APPROVED** [review]
Great job @Ramadhan0

File: src/App/Components/Articles/getArticles/Article.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    } = data;
+    const { name } = data.Category;
+    const { profileImage, firstName, username } = data.User;
+    const { comments, reads, shares } = data.stats;
+    const { allUsers, statistics } = data.rating;
+    const ratings = countRating(statistics);
+
+    const date = moment(createdAt).format('MMM Do YY');
+    const img = 'https://ielektro.es/wp-content/uploads/2017/04/ventajas-comprar-LED.jpg';
+    const defaultProImage = 'https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_960_720.png';
```
Comment (MemunaHaruna) [inline src/App/Components/Articles/getArticles/Article.js]: @Ramadhan0 what is `defaultProImage`? just use `defaultProfileImage`

File: src/App/Components/Articles/getArticles/Article.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    const articleBody = body.length > 200 ? `${body.substring(0, 210)} ...` : body;
+    const articleTitle = title.length > 52 ? `${title.substring(0, 52)}...` : title;
+    const categoryColor = (name === 'LOVE' && '#ff1744') || (name === 'TECHNOLOGY' && '#3f51b5') || (name === 'MUSIC' && '#880e4f') || (name === 'BUSINESS' && '#2196f3') || (name === 'ART' && '#9c27b0') || '#afad99';
```
Comment (MemunaHaruna) [inline src/App/Components/Articles/getArticles/Article.js:28]: @Ramadhan0 can this be placed in a constants file which can be imported and re-used?

File: src/App/Components/Common/NavProfile/navLink.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    return (
+
+      <div className="collapse navbar-collapse" id="navbarSupportedContent">
+        <ul
+          className="navbar-nav mr-auto"
+          style={{ marginLeft: '10%' }}
+        >
+          <li className="nav-item active">
+            <a className="nav-link" href="/">
+               HOME
```
Comment (MemunaHaruna) [inline src/App/Components/Common/NavProfile/navLink.js]: @Ramadhan0 why are these hard-coded?

File: src/App/_tests_/articles/article.test.js [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+import countRating from '../../../Helpers/countRating';
+
+
+const middlewares = [thunk, promiseMiddleware];
+const mockStore = configureStore(middlewares);
+
+
+describe('<articles />', () => {
+  describe('render()', () => {
+    it('should give rating', () => {
```
Comment (MemunaHaruna) [inline src/App/_tests_/articles/article.test.js]: @Ramadhan0 edit the test descriptions. Are you testing that it `should give rating` or that it `should display the right rating`?

File: src/App/_tests_/articles/articles.test.js [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+  it('Renders the component', () => {
+    wrapper.instance().componentWillReceiveProps({ articles });
+    wrapper.instance().componentWillReceiveProps({ Search });
+    jest.runAllTimers();
+    jest.useRealTimers();
+  });
+
+  it('Handle the pagination', () => {
+    wrapper.setState({ isLoading: false });
```
Comment (MemunaHaruna) [inline src/App/_tests_/articles/articles.test.js:48]: @Ramadhan0 there are no assertions here. why?

File: src/Helpers/countRating.js [hunk 8]
```diff
@@ -1,10 +1,10 @@
 export default (ratings) => {
-  let users = 0;
-  let rates = 0;
-  ratings.map((single) => {
-    rates += single.rating * single.users;
-    users += single.users;
-    return true;
-  });
-  return rates / users;
+  const x = ratings.map(item => ({
```
Comment (MemunaHaruna) [inline src/Helpers/countRating.js:2]: @Ramadhan0 @nkpremices why is this happening on the front-end?

**Review by MemunaHaruna — CHANGES_REQUESTED** [review]

File: src/App/Components/Common/NavProfile/navLink.js (hunk 5 above)
Comment (Ramadhan0) [reply src/App/Components/Common/NavProfile/navLink.js]: I wanted to add elements in navbar but @nkpremices is the one who assigned the task @MemunaHaruna

Comment (Ramadhan0) [reply src/App/Components/Common/NavProfile/navLink.js]: I going to remove this file

File: src/Helpers/countRating.js (hunk 8 above)
Comment (Ramadhan0) [reply src/Helpers/countRating.js:2]: Here I was calculating the rating percentage in order to fill the rating stars @MemunaHaruna

File: src/App/Components/Articles/getArticles/Article.js (hunk 4 above)
Comment (Ramadhan0) [reply src/App/Components/Articles/getArticles/Article.js:28]: here I am updating the class names of my div and i am the only one who needs this @MemunaHaruna

**Review by MemunaHaruna — APPROVED** [review]

File: src/App/_tests_/articles/articles.test.js (hunk 7 above)
Comment (Ramadhan0) [reply src/App/_tests_/articles/articles.test.js:48]: I resolved this

**Pull request merged**